### PR TITLE
ci: Test CMake version

### LINKED ISSUE #456: Compillation error in HPC
I was trying to compile spglib in a HPC cluster. I am getting an error related to `gtest-populate`
.
.
.
.
[ 11%] Creating directories for 'gtest-populate'
[ 22%] Performing download step (git clone) for 'gtest-populate'
-- Had to git clone more than once:
          3 times.
CMake Error at gtest-subbuild/gtest-populate-prefix/tmp/gtest-populate-gitclone.cmake:31 (message):
  Failed to clone repository: 'https://github.com/google/googletest'
gmake[2]: *** [CMakeFiles/gtest-populate.dir/build.make:102: gtest-populate-prefix/src/gtest-populate-stamp/gtest-populate-download] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/gtest-populate.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2
.
.
.
.

Can someone comment? Thanks in advance.

### PR DESCRIPTION
**Pull request by LecrisUT**
Adds tests for CMake verisons:
- Minimum supported version
- Lastest stable version
- Upcoming release candidate

### CODE REVIEW

**Comment by lan496** [comment]
The minimum required CMake version (3.25) looks too recent: for example, ubuntu 22.04 LTS installs 3.22 by default.

**Comment by LecrisUT** [comment]
Yeah, I'm holding off a bit until Ubuntu `24.04.  Debian 12 already has 3.25, and hopefully EPEL-next-9 covers RHEL based ones soon.

**Comment by LecrisUT** [comment]
CI in #457 is suffering because of CMake 3.29 failure. I will try to focus on this one a bit and at least port some basic compatibility for 3.20

File: .github/workflows/step_test.yaml [hunk 1]
```diff
@@ -84,6 +84,52 @@ jobs:
...
+  cmake-version:
+    name: >
+      🔨 CMake ${{ matrix.cmake }}
+      ${{ matrix.experimental && '[🧪 Experimental]' || '' }}
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental || false }}
+    strategy:
+      fail-fast: false
+      matrix:
+        cmake: [ "3.20", "latest", "latestrc" ]
```
Comment (lan496) [inline .github/workflows/step_test.yaml:96]: Note for me: this is minimum required CMake version

File: .github/workflows/step_test.yaml [hunk 2]
```diff
@@ -84,6 +84,52 @@ jobs:
...
+          python3 -m pip install -r requirements.txt
+      - uses: lukka/get-cmake@latest
+        with:
+          cmakeVersion: ${{ matrix.cmake }}
+      - name: Run CMake workflow
+        uses: lukka/run-cmake@v10.3
+        with:
+          workflowPreset: gcc-ci
+        continue-on-error: ${{ matrix.experimental && inputs.mask-experimental}}
+        if: matrix.cmake != '3.20'
```
Comment (lan496) [inline .github/workflows/step_test.yaml:117]: It would be nice to store "3.20" as variable.

Comment (LecrisUT) [reply .github/workflows/step_test.yaml:117]: Github worklows does not have variables except for the inputs, so at most I can move it to `ci.yaml`. But I can add a step to extract the minimum cmake version. A simple `grep | sed`

Comment (lan496) [reply .github/workflows/step_test.yaml:117]: I see. I didn't know that 😮

**Pull request merged**